source/io.perfmark/perfmark-api/Transforms/Metadata.xml

```diff
@@ -0,0 +1,8 @@
+<metadata>
+    <attr
+        path="/api/package[@name='io.perfmark']/interface[@name='StringFunction']/method[@name='apply' and count(parameter)=1 and parameter[1][@type='T']]"
+        name="return"
```

Comment:
I tried few options. Maybe I need to sit down and dive in deeper.




Following metadata 

```xml
    <attr
        path="/api/package[@name='io.perfmark']/interface[@name='StringFunction']/method[@name='apply' and count(parameter)=1 and parameter[1][@type='T']]"
        name="managedReturn"
        >
        Java.Lang.Object
    </attr>
```

causes following error:

```
./generated/io.perfmark.perfmark-api/obj/Release/net7.0-android/generated/src/Xamarin.Io.Perfmark.IStringFunction.cs(86,37): error CS1503: Argument 1: cannot convert from 'Java.Lang.Object' to 'string?' [./generated/io.perfmark.perfmark-api/io.perfmark.perfmark-api.csproj::TargetFramework=net7.0-android]
```

And this one:

```xml
    <attr
        path="/api/package[@name='io.perfmark']/interface[@name='StringFunction']/method[@name='apply' and count(parameter)=1 and parameter[1][@type='T']]"
        name="managedReturn"
        >
        string
    </attr>
```

causes:

```
./generated/io.perfmark.perfmark-api/obj/Debug/monoandroid12.0/generated/src/Xamarin.Io.Perfmark.IStringFunction.cs(19,76): error CS0738: 'IStringFunctionInvoker' does not implement interface member 'IFunction.Apply(Object?)'. 'IStringFunctionInvoker.Apply(Object)' cannot implement 'IFunction.Apply(Object?)' because it does not have the matching return type of 'Object'. [./generated/io.perfmark.perfmark-api/io.perfmark.perfmark-api.csproj::TargetFramework=MonoAndroid12.0]
./generated/io.perfmark.perfmark-api/obj/Debug/net6.0-android/generated/src/Xamarin.Io.Perfmark.IStringFunction.cs(19,76): error CS0738: 'IStringFunctionInvoker' does not implement interface member 'IFunction.Apply(Object?)'. 'IStringFunctionInvoker.Apply(Object)' cannot implement 'IFunction.Apply(Object?)' because it does not have the matching return type of 'Object'. [./generated/io.perfmark.perfmark-api/io.perfmark.perfmark-api.csproj::TargetFramework=net6.0-android]
./generated/io.perfmark.perfmark-api/obj/Debug/net7.0-android/generated/src/Xamarin.Io.Perfmark.IStringFunction.cs(19,76): error CS0738: 'IStringFunctionInvoker' does not implement interface member 'IFunction.Apply(Object?)'. 'IStringFunctionInvoker.Apply(Object)' cannot implement 'IFunction.Apply(Object?)' because it does not have the matching return type of 'Object'. [./generated/io.perfmark.perfmark-api/io.perfmark.perfmark-api.csproj::TargetFramework=net7.0-android]
```